docs/guide/database-insert-update.md

```diff
@@ -0,0 +1,159 @@
+# Inserting and updating in server-side code
+
+Apostrophe provides module methods for inserting and updating content documents in the database. These methods should be the primary tools for developers doing server-side data operations since they [emit events](/guide/server-events.md) that the CMS uses to keep all database collections updated and in sync. We will look at methods for both pieces and pages, which are similar but have some important differences.
+
+::: note
+Remember that the Apostrophe REST APIs for [pieces](/reference/api/pieces.md) and [pages](/reference/api/pages.md) are usually the best option when triggering content updates from the browser. Those API endpoints take advantage of the methods below while adding logic important to use in browsers.
+:::
+
+## Updating content documents
+
+When we have an existing piece document that to update, we use the `update()` method on the related piece type module. The `update` method takes the following arguments:
+
+| Argument | Required | Description |
+| -------- | -------- | ----------- |
+| `req` | TRUE | The associated request object. Using a provided `req` object is important for maintaining user role permissions. |
+| `piece` | TRUE | The document object that will *replace* the existing database document. |
+| `options` | FALSE | An options object, currently only used for internal draft state management. |
+
+For example, if we had a dog adoption website we may want to update `dog` pieces to indicate whether someone requested information about them. This could be after a contact form receives a submission about the dog.
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    extend: '@apostrophecms/piece-type',
+    // ...
+    methods (self) {
+      // The hypothetical contact form can only pass along the `_id` for the dog.
+      async registerInterest(dogId) {
+        const req = apos.task.getReq({ role: 'editor' });
```

Comment:
Feel like this immediately bypasses what was said earlier about the importance of using a provided req object. There's a reason (in this context the site visitor probably has no permissions), but that should be explained before we run right out and use this.

Reply:
Right. Maybe there's a better example to use here.

Reply:
Mmm. Something in a command line task would be good.